libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.Foreach.schema

```diff
@@ -28,6 +28,20 @@
                 "user.todoList"
             ]
         },
+        "index": {
+            "$role": "expression",
+            "type": "string",
+            "title": "Index property",
+            "description": "Property that holds the index of the item.",
+            "default": "dialog.foreach.index"
```

Comment:
If this is what composer uses to render as the default, then should this not be `=dialog.foreach.index` and `=dialog.foreach.value`?

Reply:
Here we just expecting a property string instead of the value of this property.

Reply:
Yes, Lu is right, this is correct.  We don't want the value, we want the property path.

---
In reply to: [405978123](https://github.com/microsoft/botbuilder-dotnet/pull/3704#discussion_r405978123) [](ancestors = 405978123)